compiler/test/dotty/tools/AnnotationsTests.scala

```diff
@@ -26,18 +26,24 @@ class AnnotationsTest:
         val defn = ctx.definitions
         val cls = requiredClass("A")
         val annotCls = requiredClass("Annot")
-        val arrayOfString = defn.ArrayType.appliedTo(List(defn.StringType))
-
-        atPhase(erasurePhase.next) {
-          val annot = cls.getAnnotation(annotCls)
-          // Even though we're forcing the annotation after erasure,
-          // the typed trees should be unerased, so the type of
-          // the annotation argument should be `arrayOfString` and
-          // not a `JavaArrayType`.
-          val arg = annot.get.argument(0).get
-          assert(arg.tpe.isInstanceOf[AppliedType] && arg.tpe =:= arrayOfString,
-            s"Argument $arg had type:\n${arg.tpe}\nbut expected type:\n$arrayOfString")
-        }
+        val arrayOfString = defn.ArrayOf(defn.StringType)
+
+        // Even though we're forcing the annotation after erasure,
+        // the typed trees should be unerased, so the type of
+        // the annotation argument should be `arrayOfString` and
+        // not a `JavaArrayType`.
+
+        val annot = atPhase(erasurePhase.next) { cls.getAnnotation(annotCls) }
+        val arg = annot.get.argument(0).get
```

Comment:
It's important to run this line at erasure, otherwise we're not forcing the annotation tree at the erasure phase. But also, why is this test suddenly failing after the change you made? They don't seem related at all, and based on the comment you added I would expect it to fail before too. If you're not sure what's going on let me know and I can try to have a look.

Reply:
I don't know why it's not failing before. But the subtyping trace and debugging tells that `Array[String] =:= Array[String]` is false after erasure due to the cause in the commit message. Just running the subtyping in an early phase is enough.

Reply:
@smarter: I've just updated the code so that we only do subtyping test before erasure. If it's green, then please have a look if you have clue why it's not failing before.

Reply:
From the investigation so far, the changes in `ClassfileParser` have changed subtyping path for `Array[String] <:< Array[String]`.

Previously, it does not touch the line `TypeComparer.compareAppliedType2`, now it reaches that line.

Reply:
After some further investigation, we know why it accidentally works before: before the change the two `Array[String]` and `Array[String]` point to the same object, thus they go through the short-cut `tp1 eq tp2` in TypeComparer, thus never reaches the line `TypeComparer.compareAppliedType2`.

The changes in ClassfileParser somehow changes the caching behavior.